python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -264,6 +339,71 @@ def unique_and_compact_node_pairs(
     # Return singleton for a homogeneous graph.
     if is_homogeneous:
         compacted_node_pairs = list(compacted_node_pairs.values())[0]
-        unique_nodes = list(unique_nodes.values())[0]
+        cat_seeds = list(cat_seeds.values())[0]
+
+    return cat_seeds, compacted_node_pairs
+
+
+def not_deduplication_compact_dst(
```

Comment:
@peizhou001 can you take a look here esp. the efficiency.

Reply:
The time complexity should be O(n). n depends on the length of dst. In order to treat the same elements as different individuals, I haven't thought of a better strategy at the moment.